l10n_nl_xaf_auditfile_export/auditfile_export.py

```diff
@@ -0,0 +1,290 @@
+        curAmntCred = lambda c, d, e: e if c > d else 0 if c < d else 0
+
+        template = self.pool.get("xml.template")
+
+        self._logger.info('create xml_template')
+        template_id = template.create(cr, uid, {'name': data.name, 'content': data.template_content})
+        try:
+
+            xml = template.generate_xml(
+                cr, uid, template_id,
+                partners=partners,
+                fiscal_year=fiscal_year,
+                date_created=datetime.date.today().strftime(_date_format),
+                software_desc=release.description,
+                software_version=release.major_version,
+                company=partners[0].company_id,
+                periods=periods,
+                accounts=accounts,
+                chartOfAccount=coa,
+                linesCount=linesCount,
+                journals=journals,
+                journals_moves=journals_moves,
+                tax_codes=tax_codes_list,
+                custSupTp=custSupTp,
+                accTp=accTp,
+                amntTp=amntTp,
+                amnt=amnt,
+                totalDebit=totalDebit,
+                totalCredit=totalCredit,
+                periodNumber=periodNumber,
+                text=text,
+                vat=vat,
+                currency=currency,
+                vatPerc=vatPerc,
+                vatAmnt=vatAmnt,
+                curAmnt=curAmnt,
+                curAmntDeb=curAmntDeb,
+                curAmntCred=curAmntCred,
+                empty="",
+            )
+
+            # validates given xml file
+            self._logger.info('validate xml')
+            _file = get_module_resource(
+                'l10n_nl_xaf_auditfile_export', 'data', 'auditfile_schema.xsd')
+            if not _file:
+                raise osv.except_osv(_('Data Error'), _('Invalid XML schema'))
+            _schema_doc = etree.parse(open(_file))
+            _schema = etree.XMLSchema(_schema_doc)
+            if not _schema.validate(xml):
+                for line in _schema.error_log:
+                    self._logger.error('%s\n' % line.message)
+                raise osv.except_osv(_('Data Error'), _('Invalid Data. Please, see logs for details.'))
+
+            # remove 'nso:' from file
+            [xml_string] = etree.tostringlist(xml)
+            xml_string = xml_string.replace("ns0:", "")
+            xml_new = etree.fromstring(xml_string, parser=None, base_url=None)
+
+            # attaches xml to module
+            template.attach_xml(
+                cr, uid, template_id, attach_to=data, xml=xml_new, name=data.name + _output_file,
+                fname=data.name + _output_file,
+                description='XML file representing this partner',
+            )
+            self._logger.info('auditfile_export created succesfully')
+            self.write(cr, uid, ids[0],
+                       {'state': 'done', 'date_printed': time.strftime('%Y-%m-%d %H:%M:%S')}, context=id)
```

Comment:
Void write to a function field 'state' without an inverse function.
Date format: import DEFAULT_SERVER_DATETIME_FORMAT from openerp.tools.